src/runtime_src/core/tools/xbutil2/OO_MemRead.cpp

```diff
@@ -132,13 +134,17 @@ OO_MemRead::execute(const SubCmdOptions& _options) const
   XBU::verbose(boost::str(boost::format("Device: %s") % xrt_core::query::pcie_bdf::to_string(xrt_core::device_query<xrt_core::query::pcie_bdf>(device))));
   XBU::verbose(boost::str(boost::format("Address: %s") % addr));
   XBU::verbose(boost::str(boost::format("Size: %s") % size));
+  XBU::verbose(boost::str(boost::format("Block count: %s") % m_count));
   XBU::verbose(boost::str(boost::format("Output file: %s") % m_outputFile));
 
   //read mem
   XBU::xclbin_lock xclbin_lock(device);
   
   try{
-    xrt_core::mem_read(device.get(), addr, size, m_outputFile);
+    for(int c = 0; c < m_count; c++) {
+      xrt_core::mem_read(device.get(), addr, size, m_outputFile);
```

Comment:
What happens if there is an address overrun?

Reply:
The function takes the smallest of the bank size or the requested read size before starting a read call. Each bank and its address are validated before it is read so this will be not be a issue.